### PR TITLE
Storage and identifier improvements

### PR DESCRIPTION
**Pull request by kzu**

### CODE REVIEW

File: Api/Http/Encoder.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+namespace NosAyudamos
+{
+    class Encoder
+    {
+        [FunctionName("encode")]
+        public IActionResult Encode([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "encoder/encode")] HttpRequest req)
+        {
+            if (!req.QueryString.HasValue ||
+                !long.TryParse(req.QueryString.Value.TrimStart('?'), out var number))
+                return new BadRequestObjectResult("Query string should contain a number to encode.");
```
Comment (pgaliano) [inline Api/Http/Encoder.cs:15]: resx?

File: Api/Http/Encoder.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+            return new OkObjectResult(Base62.Encode(number));
+        }
+
+        [FunctionName("decode")]
+        public IActionResult Decode([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "encoder/decode")] HttpRequest req)
+        {
+            if (!req.QueryString.HasValue)
+                return new BadRequestObjectResult("Query string should contain the string to decode.");
```
Comment (pgaliano) [inline Api/Http/Encoder.cs:24]: resx?

File: Core/Services/EventGridExtensions.cs [hunk 3]
```diff
@@ -31,7 +31,7 @@ public static EventGridEvent ToEventGrid(this object data, ISerializer serialize
 
             return new EventGridEvent
             {
-                Id = metadata?.EventId ?? Guid.NewGuid().ToString(),
+                Id = metadata?.EventId ?? Guid.NewGuid().ToString("n"),
```
Comment (pgaliano) [inline Core/Services/EventGridExtensions.cs:34]: PreciseTime.UtcNow.ToString("yyyy-MM-ddTHH:mm:ss.fffffffZ", CultureInfo.InvariantCulture); ?

File: Core/Services/EventGridExtensions.cs [hunk 4]
```diff
@@ -66,7 +66,7 @@ public static TableEntity ToEntity(this DomainEvent data, ISerializer serializer
 
             return new EventGridEventEntity
             {
-                RowKey = metadata?.EventId ?? Guid.NewGuid().ToString(),
+                RowKey = metadata?.EventId ?? Guid.NewGuid().ToString("n"),
```
Comment (pgaliano) [inline Core/Services/EventGridExtensions.cs:69]: PreciseTime.UtcNow.ToString("yyyy-MM-ddTHH:mm:ss.fffffffZ", CultureInfo.InvariantCulture); ?

File: Core/Services/EventGridExtensions.cs (hunk 3 above)
Comment (kzu) [reply Core/Services/EventGridExtensions.cs:34]: Yep, coming next

**Pull request merged**